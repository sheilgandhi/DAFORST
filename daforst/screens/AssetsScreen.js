import { View, Text, FlatList, StyleSheet } from 'react-native';
import React, { useEffect, useState } from 'react';
import axios from 'axios';
import Loading from '../components/Loading';
import AssetPreview from '../components/AssetPreview';

const AssetsScreen = ({ navigation }) => {
  const [assets, setAssets] = useState([]);
  const ip = '10.140.78.215'; // '192.168.68.52';

  useEffect(() => {
    // base64ToImage();
    getAssets();
  }, []);

  const getAssets = async () => {
    try {
      const response = await axios.get(`http://${ip}:3000/api/ipfs`);
      if (response) {
        setAssets(response.data);
      }
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <View>
      {assets.length > 0 ? (
        <FlatList
          testID="list"
          data={assets}
          renderItem={({ item, index }) => (
            <AssetPreview
              asset={item}
              navigation={navigation}
              key={item.key}
              index={index}
            />
          )}
        />
      ) : (
        <Loading />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
  },
  image: {
    width: 300,
    height: 300,
  },
});

export default AssetsScreen;
