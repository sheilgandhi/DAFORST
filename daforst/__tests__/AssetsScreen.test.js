import React from 'react';
import renderer from 'react-test-renderer';
import AssetsScreen from '../screens/AssetsScreen';

const navigation = {
    goBack: jest.fn(),
    navigate: jest.fn(),
    setOptions: jest.fn(),
  };

describe('AssetsScreen', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<AssetsScreen navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
