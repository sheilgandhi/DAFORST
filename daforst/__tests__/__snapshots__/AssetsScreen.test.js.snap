// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsScreen renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <ActivityIndicator
      color="#55897B"
      size="large"
    />
  </View>
</View>
`;
